community/bolt/src/main/java/org/neo4j/bolt/v2/messaging/Neo4jPackV2.java

```diff
@@ -246,7 +247,7 @@ private TimeValue unpackTime() throws IOException
         {
             long nanosOfDayUTC = unpackLong();
             int offsetSeconds = unpackInteger();
-            return time( nanosOfDayUTC, ZoneOffset.ofTotalSeconds( offsetSeconds ) );
+            return time( nanosOfDayUTC - offsetSeconds * NANOS_PER_SECOND, ZoneOffset.ofTotalSeconds( offsetSeconds ) );
```

Comment:
Why is conversion needed here?
I think bolt server should both send and receive local time with offset. Thus conversion is probably only needed in `#writeTime()` because database stores UTC.